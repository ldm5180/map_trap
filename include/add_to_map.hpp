#ifndef ADD_TO_MAP_HPP
#define ADD_TO_MAP_HPP

#include <algorithm>
#include <benchmark/benchmark.h>
#include <cstdlib>
#include <string>

static const auto add_to_map = [](const auto num_elems, auto &map) {
  // Add elements to map, this sorts automatically.
  for (auto i = 0; i < num_elems; ++i) {
    const auto random_key = std::rand();
    map[random_key] = std::to_string(map.size());
  }
  benchmark::ClobberMemory();
};

static const auto add_to_map_and_key_list = [](const auto num_elems, auto &map,
                                               auto &keys) {
  // Add elements to map, this sorts automatically.
  for (auto i = 0; i < num_elems; ++i) {
    const auto random_key = std::rand();
    map[random_key] = std::to_string(map.size());
    keys.push_back(random_key);
    std::sort(keys.begin(), keys.end());
  }
  benchmark::ClobberMemory();
};

#endif // ADD_TO_MAP_HPP
