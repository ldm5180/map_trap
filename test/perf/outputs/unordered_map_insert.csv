Run on (4 X 2594.09 MHz CPU s)
2017-10-28 16:41:43
name,iterations,real_time,cpu_time,time_unit,bytes_per_second,items_per_second,label,error_occurred,error_message
"BM_unordered_map_insert/8",168358,4932.95,4928.94,ns,,,,,
"BM_unordered_map_insert/16",116955,9903.44,9901.61,ns,,,,,
"BM_unordered_map_insert/32",52430,16292.9,16291,ns,,,,,
"BM_unordered_map_insert/64",22300,32077.5,32042.7,ns,,,,,
"BM_unordered_map_insert/128",10000,65806.1,65797.1,ns,,,,,
"BM_unordered_map_insert/256",6052,150556,150529,ns,,,,,
"BM_unordered_map_insert/512",2664,255555,255515,ns,,,,,
"BM_unordered_map_insert/1024",1000,527264,527165,ns,,,,,
"BM_unordered_map_insert/2048",843,1.16637e+06,1.16615e+06,ns,,,,,
