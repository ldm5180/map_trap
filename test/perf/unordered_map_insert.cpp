#include "add_to_map.hpp"

#include <benchmark/benchmark.h>
#include <cstdlib>
#include <unordered_map>

using map_t = std::unordered_map<int64_t, int64_t>;

static void BM_unordered_map_insert(benchmark::State &state) {
  std::srand(42);
  map_t map{};
  map.reserve(state.range(0));

  for (auto _ : state) {
    add_to_map(state.range(0), map);
  }
}
BENCHMARK(BM_unordered_map_insert)->RangeMultiplier(2)->Range(8, 8 << 8);

BENCHMARK_MAIN();
