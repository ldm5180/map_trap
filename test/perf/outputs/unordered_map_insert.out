Run on (4 X 2594.09 MHz CPU s)
2017-10-28 16:30:07
name,iterations,real_time,cpu_time,time_unit,bytes_per_second,items_per_second,label,error_occurred,error_message
"BM_unordered_map_insert/8",156705,5166.96,5160.42,ns,,,,,
"BM_unordered_map_insert/16",128455,10005,10009.1,ns,,,,,
"BM_unordered_map_insert/32",52414,16645.5,16653.3,ns,,,,,
"BM_unordered_map_insert/64",22016,32339.3,32314.4,ns,,,,,
"BM_unordered_map_insert/128",10000,67395.8,67427.6,ns,,,,,
"BM_unordered_map_insert/256",6084,155844,155905,ns,,,,,
"BM_unordered_map_insert/512",2840,309440,309591,ns,,,,,
"BM_unordered_map_insert/1024",1000,539238,538076,ns,,,,,
"BM_unordered_map_insert/2048",752,1.19958e+06,1.20011e+06,ns,,,,,
